mirage/tls_mirage.ml

```diff
@@ -37,16 +37,10 @@ module Make (F : Mirage_flow.S) = struct
     | Ok (`Data _ | `Eof as x) -> x
     | Error e                  -> `Error (`Read e)
 
-  let lift_write_result = function
-    | Ok ()   -> `Ok ()
-    | Error e -> `Error (`Write e)
-
   let check_write flow f_res =
-    let res = lift_write_result f_res in
-    ( match flow.state, res with
-      | `Active _, (`Eof | `Error _ as e) ->
-          flow.state <- e ; FLOW.close flow.flow
```

Comment:
I think this diff is a part for the `shutdown` PR, I'm right?

Reply:
no -- well, the purpose here is to prepare for shutdown. A first step is to not `close` the underlying flow on arbitrary errors -- so the caller has to close the flow on error conditions.